### PR TITLE
f-footer@2.0.0-beta.7 - Country selector component

### PR DESCRIPTION
**Pull request by xander-marjoram**
### Added
- `CountrySelector` component logic and styles
- `CountrySelector` tests

### CODE REVIEW

File: packages/f-footer/src/components/LinkList.vue [hunk 1]
```diff
@@ -151,8 +151,7 @@ export default {
             width: auto;
         }
 
-        &:hover,
-        &:focus {
```
Comment (dandel10n) [inline packages/f-footer/src/components/LinkList.vue:155]: Why you removed focus from here?

File: packages/f-footer/src/components/CountrySelector.vue [hunk 2]
```diff
@@ -1,19 +1,151 @@
...
-        <p>Country selector component</p>
+    <div 
+        class="c-footer-countrySelector"
+        @keyup.esc="hideCountryList">
+        <button
+            id="countrySelector-button"
+            class="c-countrySelector-link c-countrySelector-button"
+            type="button"
+            @click="toggleCountryList"
+            :aria-expanded="showCountryList ? 'true' : 'false'">
```
Comment (dandel10n) [inline packages/f-footer/src/components/CountrySelector.vue]: I would recommend to add `:aria-controls` attribute to show relationship of the button with the expandable region. The list of flags will need an id.

File: packages/f-footer/src/components/CountrySelector.vue [hunk 3]
```diff
@@ -1,19 +1,151 @@
...
+            <flag-icon 
+                :country-code="currentCountryKey" />
+            {{ currentCountryName }}
+            <chevron-icon
+                v-if="!showCountryList"
+                :is-small="true" />
+            <cross-icon v-else />
+        </button>
+        <ul
```
Comment (dandel10n) [inline packages/f-footer/src/components/CountrySelector.vue:23]: and `:aria-labelledby="countrySelector-button"` and `role="region"` here

**Comment by dandel10n** [comment]
Will jest tests come in separate PR for this component?

File: packages/f-footer/src/components/LinkList.vue (hunk 1 above)
Comment (xander-marjoram) [reply packages/f-footer/src/components/LinkList.vue:155]: On focus, we already have the outline, we don't need the underline as well. This matches what we currently have in production (although the outline colour is different)

File: packages/f-footer/src/components/CountrySelector.vue [hunk 4]
```diff
@@ -1,19 +1,150 @@
 <template>
-    <div class="c-footer-countrySelector">
-        <p>Country selector component</p>
+    <div 
+        class="c-footer-countrySelector"
```
Comment (ashleynolan) [inline packages/f-footer/src/components/CountrySelector.vue]: Could probably just call this `c-countrySelector`?  There's likely no dependency on it being a part of the footer and that way we could pull it out of the footer later if we ever wanted to.

File: packages/f-footer/src/components/CountrySelector.vue [hunk 5]
```diff
@@ -1,19 +1,150 @@
...
+    padding: spacing(x2);
+    display: flex;
+    align-items: center;
+    justify-content: space-around;
+    background-color: $grey--lightest;
+    border: none;
+    cursor: pointer;
+}
+
+.c-footer-countrySelector-list {
```
Comment (ashleynolan) [inline packages/f-footer/src/components/CountrySelector.vue]: `c-countrySelector-list`?

**Review by ashleynolan — CHANGES_REQUESTED** [review]
Couple of small updates from me

**Comment by ashleynolan** [comment]
@xander-marjoram Needs a rebase and update to the description.

File: packages/f-footer/src/components/CountrySelector.vue (hunk 4 above)
Comment (dandel10n) [reply packages/f-footer/src/components/CountrySelector.vue]: ✅

**Review by ashleynolan — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Pull request merged**